### PR TITLE
fix(ci): stage CLI releases through npm approval

### PR DESCRIPTION
**Pull request by WcaleNieWolny**
## Summary (AI generated)

- replace direct `bun publish` with `npm stage publish` for stable (`latest`) and alpha (`next`) CLI releases
- dispatch `npm-stage-approve` to `Cap-go/automations` from a downstream job before GitHub release creation
- keep a successful npm stage isolated so failed approval dispatches can be retried without restaging the same version
- add a focused workflow contract test covering registry setup, credentials, tags, job boundaries, retry safety, dispatch payload, and step ordering

## Motivation (AI generated)

The organization token renewer now creates a short-lived token for npm staged publishing. Direct `bun publish` falls into interactive npm authentication and fails, while retrying a combined stage-and-dispatch job can conflict with an already pending package version.

## Business Impact (AI generated)

This restores automated CLI releases while preserving npm's private WebAuthn approval boundary and gives operators a safe failed-job retry path.

Deployment prerequisite: the organization Actions secret `NPM_STAGE_DISPATCH_TOKEN` must be made available to the public `Cap-go/capgo.app` repository before the next CLI tag release. Its current **Private repositories** visibility does not expose it to this workflow.

## Test Plan (AI generated)

- `bunx vitest run tests/release-scope.test.ts` (13 passed)
- `bun test:unit` (2,209 passed)
- `bunx eslint tests/release-scope.test.ts .github/workflows/publish_cli.yml`
- `bun lint` (0 errors; 38 pre-existing warnings)
- `git diff --check origin/main...HEAD`

## Summary by CodeRabbit

- **New Features**
  - CLI releases now use staged npm publishing, with stable releases on `latest` and alpha releases on `next`.
  - Releases include an approval step before publication is finalized.
  - Changelog links and comparisons now reflect the appropriate release channel.

- **Bug Fixes**
  - Improved publishing authentication and release workflow reliability.

- **Documentation**
  - Added guidance for staged CLI publishing and release approval.

- **Tests**
  - Expanded coverage for publishing channels, authentication, approvals, and release ordering.

### CODE REVIEW

File: .github/workflows/publish_cli.yml [hunk 1]
```diff
@@ -64,16 +65,27 @@ jobs:
...
         if: ${{ contains(github.ref, '-alpha.') }}
+        working-directory: cli
         env:
-          NPM_CONFIG_TOKEN: ${{ secrets.NPM_TOKEN }}
-        run: bun publish --cwd cli --tag next --access public
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: npm stage publish --access public --tag next
+      - name: Request npm stage approval
+        env:
+          GH_TOKEN: ${{ secrets.NPM_STAGE_DISPATCH_TOKEN }}
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply .github/workflows/publish_cli.yml:100]: Confirmed. Organization secret visibility is an external deployment prerequisite, not a repository-code setting. It is called out in the design specification and PR body: `NPM_STAGE_DISPATCH_TOKEN` must include public repository `Cap-go/capgo.app` before the next CLI tag. The dispatch step intentionally fails closed if that prerequisite is missing.
